add CEDAR and ARCHS4 and overhaul and simplify code base (#28)
jhudsl/ITCR_Tables: README.md

# ITCR_Tables

## General description of repo

> This repo uses the [OTTR template](https://github.com/jhudsl/OTTR_Template) to [build the ITCR tools/resources table website](https://jhudatascience.org/ITCR_Tables/index.html). There are 6 tables that are generated (resources, clinical tools, imaging tools, multi-data type tools, omics tools, and all tools) using 5 input files (resources, clinical, imaging, multi-data, and omics). The resources table needs to be generated first so that a file with identifier acronyms is written that can be used in building the rest of the tables. Within each Rmarkdown building the resource, clinical, imaging, multi-data, or omics tables, the input file is read-in and then the table is formatted to add hyperlinks, rearrange columns, etc. Rather than saving these modified tables (clinical, imaging, multi-data, and omics) as output files, `knitr::knit_child` is used to force rendering the clinical, imaging, multi-data, and omics Rmarkdowns within the `allTables.Rmd`. This then allows the `allTables.Rmd` to inherit the modified table variables. Since the resources table has to be generated first, `knitr::knit_child` is used with the `resourcesTable.Rmd` first. The rows of these tables are joined together to form the all tools table (resources are not included in this). Two tools that were listed twice (each were listed once as multi-data type and once as omics) in this combo table were de-duplicated. If you need to add or edit existing resources/tools, see the instructions below. Which file you edit, depends on the type of tool or resource that you are adding/editing.      

Reviewer: ```suggestion
# Overview

This repo uses the [OTTR template](https://github.com/jhudsl/OTTR_Template) to [build the ITCR tools/resources table website](https://jhudatascience.org/ITCR_Tables/index.html). 

There are 6 tables that are generated:

- data resources
- clinical tools
- imaging tools
- multi-data type tools
- omics tools
- and all tools

These are generated using 5 input files (data resources, clinical, imaging, multi-data, and omics). The data resources table needs to be generated first so that a file with identifier acronyms is written that can be used in building the rest of the tables. Within each R Markdown file building the tables, the input file is read-in and then the table is formatted to add hyperlinks, rearrange columns, etc. 

Rather than saving these modified tables (clinical, imaging, multi-data, and omics) as output files, `knitr::knit_child` is used to force rendering the clinical, imaging, multi-data, and omics R Markdowns within the `allTables.Rmd`. This then allows the `allTables.Rmd` to inherit the modified table variables. Since the resources table has to be generated first, `knitr::knit_child` is used with the `resourcesTable.Rmd` first. The rows of these tables are joined together to form the all tools table (data resources are not included in this). 
      
```

Reviewer: regarding "The rows of these tables are joined together to form the all tools table (resources are not included in this)", maybe I am missing something but it seems like the current version shows the data resources available for the all tools table: https://jhudatascience.org/ITCR_Tables/allTables.html

Reviewer: also maybe this can just be part of the PR message: Two tools that were listed twice (each were listed once as multi-data type and once as omics) in this combo table were de-duplicated.  (maybe it already is)

Reviewer: Overall though this is awesome!

Reviewer: I think we can remove the contributing info since we say it later

## Adding to or editing the Resource Table

> The resource table has different information for each resource compared to the tools in the other tables.

Reviewer: ```suggestion
The data resource table has additional information about data that can be accessed by the various computing platform tools in the other tables.
```

* [`omicsTable.Rmd`](https://github.com/jhudsl/ITCR_Tables/blob/main/omicsTable.Rmd) -- renders the ["Computing Resources --> Omics Platforms" page](https://jhudatascience.org/ITCR_Tables/omicsTable.html)
* [`allTables.Rmd`](https://github.com/jhudsl/ITCR_Tables/blob/main/allTables.Rmd) -- renders the ["Computing Resources --> All platforms" page](https://jhudatascience.org/ITCR_Tables/allTables.html)
* [`contact.Rmd`](https://github.com/jhudsl/ITCR_Tables/blob/main/contact.Rmd) -- renders the ["Contact" page](https://jhudatascience.org/ITCR_Tables/contact.html)
* [`feedback.Rmd`](https://github.com/jhudsl/ITCR_Tables/blob/main/feedback.Rmd) -- renders the ["Feedback Form" page](https://jhudatascience.org/ITCR_Tables/feedback.html)
* [`index.Rmd`](https://github.com/jhudsl/ITCR_Tables/blob/main/index.Rmd) -- renders the [home page](https://jhudatascience.org/ITCR_Tables/index.html)
* [`platformsOverview.Rmd`](https://github.com/jhudsl/ITCR_Tables/blob/main/platformsOverview.Rmd) -- renders the ["Computing Resources --> Platforms overview" page](https://jhudatascience.org/ITCR_Tables/platformsOverview.html)

### Other files, not rendered

> * [`scripts/format-tables.R`](https://github.com/jhudsl/ITCR_Tables/blob/main/format-tables.R) -- functions that can be used to format the tables for the various `Table.Rmd` files. This script is sourced within the Rmarkdown files after loading libraries so the functions are available.

Reviewer: ```suggestion
* [`scripts/format-tables.R`](https://github.com/jhudsl/ITCR_Tables/blob/main/format-tables.R) -- functions that can be used to format the tables for the various `Table.Rmd` files. This script is sourced within the R Markdown files after loading libraries so the functions are available.
```
